src/imgui/Shortcuts.cc

```diff
@@ -0,0 +1,60 @@
+#include "Shortcuts.hh"
+#include "imgui_internal.h"
+
+#include "SettingsConfig.hh"
+
+#include <array>
+
+namespace openmsx {
+
+static constexpr auto defaultShortcuts = std::array{
+	Shortcuts::Data{ShortcutIndex::GOTO_MEMORY_ADDRESS, ImGuiMod_Ctrl | ImGuiKey_G, ShortcutType::LOCAL},
+	Shortcuts::Data{ShortcutIndex::GOTO_DISASM_ADDRESS, ImGuiMod_None, ShortcutType::LOCAL},
+};
+
+Shortcuts::Shortcuts(SettingsConfig& config_)
+	: config(config_)
+	, shortcuts{}
+{
+	config.setShortcuts(*this);
+	loadInit();
+}
+
+void Shortcuts::loadInit()
```

Comment:
![image](https://github.com/openMSX/openMSX/assets/7815819/b190ec5f-4fe3-41b6-9ea8-75b541d9ffe3)
The "Clear" button in the shortcut capturing modal window already restores individual shortcuts.

Reply:
Euhm, no. At least if I understood the code correctly then "Clear" will remove the shortcut (set it to `ImGuiKey_None`). That's also a useful operation, but it's different from restoring the default binding.

For example I can change the shortcut for `GOTO_MEMORY_ADDRESS` from `CTRL+G` to e.g. `CTRL+A`. If I then press the "Clear" button i remove this shortcut (no key-combination will activate this function). But it may also be useful to restore the original shortcut (`CTRL+G`). Because as a user I may no longer remember what the original shortcut was.

Now that I think about this: it might be useful, during saving the shortcuts to `settings.xml`, to omit the shortcuts that still have their default settings. This will make the `settings.xml` file a lot shorter, and it's useful to upgrade to newer openMSX versions: then you more easily adapt to changes in default shortcuts in newer openMSX versions. (And we have to make sure that a cleared shortcut can be distinguished from a default shortcut).

Reply:
A third button for default short cut mapping:
![image](https://github.com/openMSX/openMSX/assets/7815819/803a4229-91be-4c31-b274-3ee6edce8312)